WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsViewModel.kt

```diff
@@ -33,32 +35,28 @@ class StatsViewModel
     private val _showSnackbarMessage = SingleLiveEvent<SnackbarMessageHolder>()
     val showSnackbarMessage: LiveData<SnackbarMessageHolder> = _showSnackbarMessage
 
-    private var _site: SiteModel? = null
-    val site: SiteModel
-        get() = checkNotNull(_site) { "Trying to access unitialized site" }
-
-    private var currentStatsType = StatsListType.INSIGHTS
-
     fun start(site: SiteModel) {
         // Check if VM is not already initialized
-        if (_site == null) {
-            _site = site
+        if (!isInitialized) {
+            isInitialized = true
 
-            loadStats()
+            this.site = site
+            this.insightsViewModel.reset()
+
+            uiScope.loadStats()
         }
     }
 
-    private fun loadStats() = defaultScope.launch {
-        val loadState = FETCHING
-        reloadStats(loadState)
-
-        isInitialized = true
+    private fun CoroutineScope.loadStats() = launch {
+        reloadStats()
     }
 
-    private suspend fun reloadStats(state: StatsListState = FETCHING) {
-        _listState.setOnUi(state)
+    private suspend fun reloadStats() {
+        _listState.value = FETCHING
+
+        insightsViewModel.loadInsightItems(site)
```

Comment:
I think we should be calling a method like `onRefresh(site)` instead. This way we can handle the force refresh logic in the ViewModel. It could actually be one of the abstract methods on the (future) base view model.

Reply:
I'll add the in the ScopedViewModel PR because the `CoroutineScope` usage changed.